### ISSUE TITLE
Encountering dll not found exception using mono

### ISSUE THREAD

**Issue by jharonfe** [pythonnet/pythonnet#985]
### Environment

-   Pythonnet version: 2.3.0
-   Python version: 3.6.8
-   Operating System: Ubuntu 18.04.3
-   Mono: 6.4.0.198

### Details

Attempting to port .NET application from windows to linux running under mono. This application has a dependency on the `Python.Runtime.dll` version number above.

The build of the program in the code below encounters a `DLLNotFoundException`.
```csharp
    public static void Main(string[] args)
        {
            PythonEngine.Initialize();
            using (Py.GIL())
            {
                PythonEngine.Exec("print('hello')");
            }

            Console.WriteLine("done");
            Console.ReadLine();
        }
```

I've tested variations of the code above that include setting the `PYTHONHOME` and `PATH` variables. But I still encounter the same exception.

See the full exception message, stack trace below. 
```
System.DllNotFoundException: python35 assembly:<unknown assembly> type:<unknown type> member:(null)
  at (wrapper managed-to-native) Python.Runtime.Runtime.Py_IsInitialized()
  at Python.Runtime.Runtime.Initialize () [0x00001] in <11c81f5f20364a28b963bcd102f70399>:0 
  at Python.Runtime.PythonEngine.Initialize (System.Collections.Generic.IEnumerable`1[T] args, System.Boolean setSysArgv) [0x0001b] in <11c81f5f20364a28b963bcd102f70399>:0 
  at Python.Runtime.PythonEngine.Initialize (System.Boolean setSysArgv) [0x00006] in <11c81f5f20364a28b963bcd102f70399>:0 
  at Python.Runtime.PythonEngine.Initialize () [0x00001] in <11c81f5f20364a28b963bcd102f70399>:0 
  at MonoPythonnet.MainClass.Main (System.String[] args) [0x00001] in /home/jon/ws/2019.11.09.test.MonoPythonnet/MonoPythonnet/Program.cs:11 
[ERROR] FATAL UNHANDLED EXCEPTION: System.DllNotFoundException: python35 assembly:<unknown assembly> type:<unknown type> member:(null)
  at (wrapper managed-to-native) Python.Runtime.Runtime.Py_IsInitialized()
  at Python.Runtime.Runtime.Initialize () [0x00001] in <11c81f5f20364a28b963bcd102f70399>:0 
  at Python.Runtime.PythonEngine.Initialize (System.Collections.Generic.IEnumerable`1[T] args, System.Boolean setSysArgv) [0x0001b] in <11c81f5f20364a28b963bcd102f70399>:0 
  at Python.Runtime.PythonEngine.Initialize (System.Boolean setSysArgv) [0x00006] in <11c81f5f20364a28b963bcd102f70399>:0 
  at Python.Runtime.PythonEngine.Initialize () [0x00001] in <11c81f5f20364a28b963bcd102f70399>:0 
  at MonoPythonnet.MainClass.Main (System.String[] args) [0x00001] in /home/jon/ws/2019.11.09.test.MonoPythonnet/MonoPythonnet/Program.cs:11 
```
I also have included a zip of the output of the mono execution with log level set to debug.
[info.log.zip](https://github.com/pythonnet/pythonnet/files/3827868/info.log.zip)

Some help sorting through this would be greatly appreciated. Thanks.

**Comment by lostmsu** [comment]
`pythonnet` packages on NuGet are built for Windows. You need to install Python.NET into you Python with `pip` (e.g. https://pypi.org/project/pythonnet/), and also have `python3.5m.so` in your `PATH`.

**Comment by jharonfe** [comment]
Thanks for the feedback. I installed `pythonnet` 2.4.0. I also added `python3.5m.so` to the environment path. Unfortunately I still encounter the same `DLLNotFoundException` for the `Python.Runtime.dll` at the same spot when running the build of the test program.

The `Python.Runtime.dll` that's used was sourced from NuGet and you say "are built for Windows". Are there any binaries on NuGet that are compatible with mono ? 

I cloned a local copy of the `pythonnet.git` repo. I see that the solutions there contain build options for mono. I was going to generate a local build of the `Python.Runtime` project. However when these solution files are opened using monodevelop on ubuntu, the `Python.Runtime` project does not load with a message "Invalid configuration mapping". Do you have any suggestions how to load this project using monodevelop ?

**Comment by lostmsu** [comment]
When you install Python.NET from pip, it also installs an appropriate version of `Python.Runtime.dll` somewhere in Python directories, likely `site`.

**Comment by jharonfe** [comment]
I tested this last night. Here is a quick summary of what happened:

1. After I replaced the `Python.Runtime.dll` with the version from the `pythonnet` installation I ran the test app again. I no longer encounter the `DLLNotFoundException`. Unfortunately I encountered a new issue, `EntryPointNotFoundException`.
2. I reviewed some of the issues in pythonnet for this exception and there were several that were relevant: #245, #303, #740, et al. I tried to implement the solution described in #303 but without success, perhaps in part due to the fact that I could not locate some of the artifacts described such as `Python.Runtime.dll.config` and `clr.so`.
3. In the course of reviewing #303 I discovered QuantConnect/Lean has a fork of pythonnet. See [here](https://github.com/QuantConnect/Lean/tree/master/Algorithm.Python#quantconnect-python-algorithm-project). In addition they maintain a set of builds of the `Python.Runtime` project on nuget. See [here](https://www.nuget.org/packages/QuantConnect.pythonnet/). I tested the linux version if this `dll` and was able to get the test program to run under mono.
